### PR TITLE
C#: Improve diagnostic messages and add help link

### PR DESCRIPTION
**Pull request by raulsntos**
- Reword diagnostic messages.
- Add help link to diagnostics.
	- Depends on https://github.com/godotengine/godot-docs/pull/7781.

### CODE REVIEW

File: modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/Common.cs [hunk 1]
```diff
@@ -7,16 +7,18 @@ namespace Godot.SourceGenerators
 {
     public static class Common
     {
+        private const string helpLinkFormat = "https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/diagnostics/{0}.html";
```
Comment (raulsntos) [inline modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/Common.cs:10]: We should probably use the specific Godot version here instead of _stable_.

**Comment by mhilbrunner** [comment]
We merged the docs counterpart PR.

File: modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/Common.cs (hunk 1 above)
Comment (YuriSizov) [reply modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/Common.cs:10]: It's available at build time in `methods.py`, or at compile time as a `VERSION_DOCS_URL` define in C++.

**Review by YuriSizov — APPROVED** [review]
I agree that you should de-hardcode the link base, but LGTM otherwise.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**